### ISSUE TITLE
Add/Edit Connection in REST API should reject non-JSON "extra" field

### ISSUE THREAD

**Issue by bbovenzi** [apache/airflow#38163]
### Apache Airflow version

main (development)

### What happened?

When creating or updating a connection via the rest api. We do not run any validation against `extra`. While from the UI we we do enforce that extra is json with 
```
The Extra connection field contained an invalid value for Conn ID: test.

If connection parameters need to be added to Extra, please make sure they are in the form of a single, valid JSON object.


The following Extra parameters were not added to the connection:
{invalid json}
```

### What you think should happen instead?

We should apply the same validation in the REST API code and throw an error if `extra` is not valid json

### How to reproduce

Make a connection with the REST API

### Operating System

all

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by bbovenzi** [comment]
Actually, we only accept a string. We should change it to be a real object.

<img width="709" alt="Screenshot 2024-03-14 at 2 50 16 PM" src="https://github.com/apache/airflow/assets/4600967/6929537f-bf4e-494f-9591-53a7dd3ebd3c">

**Comment by harsh-doshii** [comment]
Hi @bbovenzi can you please assign this to me? This looks like a good first issue that I can work on. Thank you!

**Comment by harsh-doshii** [comment]
Made and tested some changes locally, attaching screenshots. 
Case 1: Passing a string - gives 400 bad request
Case 2: Passing an object - doesn't give a bad request
<img width="1198" alt="image" src="https://github.com/apache/airflow/assets/54910290/94cde9fe-f091-4412-9f43-bb59e9b3652c">
<img width="1135" alt="image" src="https://github.com/apache/airflow/assets/54910290/d7e32ab7-7c66-40aa-ab19-e5606581a15a">

Do let me know your thoughts. I will raise a PR soon!

**Comment by potiuk** [comment]
Raise a PR -> then we can have meaningful discussion seeing the code :)

**Comment by Taragolis** [comment]
I'm not sure could we change this behavior before Airflow "3.0". Internally Connection.extra supports just a string object, and it marked mandatory to be a JSON object into Airflow 3.0

https://github.com/apache/airflow/blob/b610c58a7b3c85995ad72beeac5a8dbdce10e7cf/airflow/models/connection.py#L184-L209

**Comment by Bowrna** [comment]
@Taragolis I tried to add the connections from CLI with invalid JSON value in extra param in `main` branch Airflow and it got added without throwing any error.
<img width="617" alt="Screenshot 2024-03-27 at 11 29 31 AM" src="https://github.com/apache/airflow/assets/10162465/bd534147-1784-4bc8-91ca-5336ddd5132a">

**Comment by Bowrna** [comment]
@Taragolis I think issue specifically occurs if we are creating connection via JSON format.

https://github.com/apache/airflow/blob/a03324382b2f891fbcecf5ff637213a08e9b877b/airflow/cli/commands/connection_command.py#L250-L271

Here you see that conn_uri and else case part of the code validates the extra param when `set_extra` function is invoked. But for the case of conn-json, the extra part in the param is not validated.

**Comment by bbovenzi** [comment]
Yes, instead of changing it to an object in the REST API. We should confirm that the string can be converted to valid JSON.

**Comment by pierrejeambrun** [comment]
This is still relevant and need to be fixed in AF3

**Comment by pierrejeambrun** [comment]
Closing, fixed by https://github.com/apache/airflow/pull/54034
